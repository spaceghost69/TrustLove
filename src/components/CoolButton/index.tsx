import React from 'react';

import "./index.css";

/*

primary'
'secondary'
'success'
'danger'
'warning'
'info'
'light'
'dark'

*/
interface CoolButtonProps {
  children: any;
  variant: string;
  onClick: any;
}

const CoolButton: React.FC<CoolButtonProps> = ({ children, variant, onClick }) => {
  
  return(
    <button
        onClick={(event) => onClick(event)}
        className={variant}
    >{children}</button>
  )
}

export default CoolButton;