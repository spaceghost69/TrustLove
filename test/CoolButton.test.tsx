import React from "react";
import renderer from "react-test-renderer";
import CoolButton from "../src/components/CoolButton";

describe("Snapshot testing of CoolButton", () => { 
    it("CoolButton snapshot", () => { 
        const CoolButtonS = renderer.
        create(<CoolButton children={undefined} variant={""} onClick={undefined} />).toJSON(); 
        expect(CoolButtonS).toMatchSnapshot(); 
    }); 
});