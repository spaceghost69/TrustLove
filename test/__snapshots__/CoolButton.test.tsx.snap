// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing of CoolButton CoolButton snapshot 1`] = `
<button
  className=""
  onClick={[Function]}
/>
`;
